libs/net/vmspec.py

```diff
@@ -20,40 +23,64 @@ class VMInterfaceStatusNotFoundError(Exception):
     pass
 
 
-def lookup_iface_status(vm: BaseVirtualMachine, iface_name: str) -> ResourceField:
+class VMInterfaceStatusStillExistsError(Exception):
+    pass
+
+
+def _default_interface_predicate(interface: ResourceField) -> bool:
+    return "guest-agent" in interface["infoSource"] and interface[IP_ADDRESS]
```

Comment:
Are you sure that having an IP address is a condition that the interface must follow?